### DISCUSSION TITLE
Nextra: docs generator for Next.js & more

### DISCUSSION THREAD

**Discussion by timneutkens** [vercel/next.js#19734] [Show and tell]
- Zero-configuration
- Optimized for MDX
- Filesystem conventions 
- Retains all Next.js power 
- Friendly to Next.js version bumps 

https://twitter.com/rauchg/status/1333137455176851457

**Comment by mewwts** [comment]
How does Nextra play with an existing Next.js codebase? Would it be easy to integrate the docs into existing pages in the Next.js app?

**Reply by shuding** [reply]
@mewwts It won't affect your existing code if you don't have existing `.md` or `.mdx` pages inside your /pages directory. It will be easy to integrate and you can move all your docs into a /pages/docs folder (or other directory names). 

You can think of Nextra as a loader, which renders Next.js `.md` and `.mdx` files with the docs theme.

**Reply by mewwts** [reply]
Thanks @shuding! Could I e.g. in pages/feature1/index.tsx import portions of docs from pages/docs/feature1.mdx and display _within_ the feature?

**Reply by leonidkuznetsov18** [reply]
@shuding Could you please explain, how I can use nextra with nextjs 13 and use only /app directory?
Right now I have 3 issues:
 - cant use only one directory /app, need create /app and /pages
 - if I need for example create documentation for /docs route and use on this page nextra only I cant change tree structure because in /pages. I need also create /docs folder.
 - have warning "warn Duplicate page detected. pages/docs/writer.mdx and pages/docs/about.mdx both resolve to /docs/about"

**Comment by rjoaopereira** [comment]
@timneutkens Is [dokz](https://github.com/remorses/dokz) a valid option also in your opinion?
I'm not associated with it in any way.

**Comment by t0t07** [comment]
Is there a roadmap for the project? I am very interested in using it to build my personal site

**Reply by shuding** [reply]
All the currently open issues are something we'd love to tackle/support.

**Comment by tbecerrp** [comment]
Hi, I can't implementation nextra with productive app with next.  
I want something like this: 
```
-pages
        -Docs
            -doc.md
        -_app.tsx
        
```

**Comment by theroyakash** [comment]
Can we add Latex math blocks?

**Comment by ghost** [comment]
Great job with Nextra!

But how can I modify the basepath? Like "<server url>/documentation"?

**Reply by leodr** [reply]
If you're talking about hosting on GitHub pages for example where your entire application is on a base path use the `basePath` option: https://nextjs.org/docs/api-reference/next.config.js/basepath

If you want Nextra to do your docs site under `/docs` and have a regular Next.js application on the base path just create a `docs` folder in your `pages` directory and fill it with your markdown files. Nextra will then only take control of that folder/path.

**Comment by sohamsshah** [comment]
How can I contribute in development of `Nextra`? Already Excited for this feature
